app/controllers/forms_controller.rb

```diff
@@ -41,8 +41,17 @@ def calculate_early_result
     if last_tag_in_group?(:gross_income)
       cfe_result = CfeService.result(session_data, Steps::Helper.completed_steps_for(session_data, step))
       session_data["early_result"] = { "result" => cfe_result.gross_income_result,
-                                       "gross_income_excess" => cfe_result.gross_income_excess }
-      # add analytics tracking here? completed user journeys
+                                       "gross_income_excess" => cfe_result.gross_income_excess,
+                                       "type" => "gross_income" } # I think we have to set this here so we have it in the session, when calling the `JourneyLoggerService`
+      track_completed_journey_for_early_result
     end
   end
+
+  def track_completed_journey_for_early_result
+    session_data["api_response"] = CfeService.call(session_data, Steps::Helper.completed_steps_for(session_data, step))
+    calculation_result = CalculationResult.new(session_data)
+    office_code = signed_in? && current_provider.present? ? current_provider.first_office_code : nil
+    assessment_id_with_early_result_suffix = "#{assessment_id}_early_result" # This is so we do not overrite newly created early result entry, if an user decides to carry on with check or change answers
```

Comment:
to be honest I'm not sold yet on using a suffix, but I can't really say exactly why, it just feels really hacky lol - have you dug into whether the original tracking method could be extended to have some conditionals/guarding in to prevent early results from updating?

Reply:
I did dig into this, but very happy to pair on making this bit in the `JourneyLoggerService` guard for early results

>       CompletedUserJourney.transaction do
>         if (journey = CompletedUserJourney.find_by(assessment_id:))
>           journey.update!(attributes)
>         else
>           CompletedUserJourney.create!(attributes.merge(assessment_id:))
>         end
>       end

What I tried checking was if there was an `assessment_id:` which had `early_result_type` as not `nil`. The problem I ran into, when I tried that, was some weird thing with caching of what to put into `CompletedUserJourney`. Lets pair on this if you're happy to.

Having the suffix means that, as per the ticket, we always create an entry into `CompletedUserJourney` when there is an early ineligible result and never update it. 

I had the opposite vibe of this being simpler to understand rather then delving into more if conditionals. 🤣

Reply:
I had this monstrosity which gave me bad vibes:

>       CompletedUserJourney.transaction do
>         early_ineligible_result_journey = CompletedUserJourney.find_by(assessment_id:) && CompletedUserJourney.where(assessment_id:).where.not(early_result_type: nil).exists?
>         if (journey = CompletedUserJourney.find_by(assessment_id:))
>           journey.update!(attributes) unless early_ineligible_result_journey
>         else
>           CompletedUserJourney.create!(attributes.merge(assessment_id:))
>         end
>       end

Reply:
Maz and I paired on doing this an alternative way without the suffix but we ran into other issues - firstly turns out the DB field has a uniqueness constraint on assessment_id, which would mean we'd have to change that if we wanted to store it potentially more than once. Also, having a more complex tracking event that is for both early tracking and end of service tracking creates issues when a user goes back and forth and triggers tracking multiple times. Suffix seems safest option.

Reply:
Ok actually we could do something like this:

```
 CompletedUserJourney.transaction do
        scope = CompletedUserJourney.where(assessment_id:)

        journey = if check.early_result_type.present?
                    scope.where(early_result_type: check.early_result_type).first
                  else
                    scope.where(early_result_type: nil).first
                  end

        if journey
          journey.update!(attributes)
        else
          CompletedUserJourney.create!(attributes.merge(assessment_id:))
        end
      end
```

Reply:
We get 'duplicate key value violations' if we go with the above instead the suffix.

I think we're in the unavoidable loop of having an existing row in `CompletedUserJourney` and then trying to create a new one which shares the same `assessment_id`. 

It's a sensible constraint of the schema, solved by the suffix.